geoutils/georaster/raster.py

```diff
@@ -1136,6 +1136,13 @@ def reproject(
             dst_nodata = self.nodata
             if dst_nodata is None:
                 dst_nodata = _default_ndv(dtype)
+                # if dst_nodata is already being used, raise a warning.
+                # TODO: for uint8, if all values are used, apply rio.warp to mask to identify invalid values
+                warnings.warn(
+                    f"For reprojection, dst_nodata must be set. Default chosen value {dst_nodata} exist in \
+self.data. This may have unexpected consequences. Consider setting a different nodata with \
```

Comment:
Did you mean to write "Default chosen value **might** exist in self.data", or is there a test to add here before triggering the warning?

Reply:
Good catch, I think I forgot to add an if statement!

Reply:
Fixed in f3f5f7a